.github/workflows/sync_wiki.yml

```
@@ -0,0 +1,75 @@
+name: Synchronize wiki pages
+
+################################################################################
+#
+#   🚨  🚨  🚨
+#
+#   If you edit this file, be sure to make the same edit in all of the other
+#   repos whose wikis should participate in synchronization. The GitHub App we
+#   use to synchronize the repos themselves does not have permission to sync
+#   workflow files, so this file (and others in .github/workflows) must be
+#   manually copied.
+#
+################################################################################
+
+on:
+  gollum:
+
+jobs:
+  synchronize:
+    strategy:
+      matrix:
+        # The list of wikis to be synced. A GitHub wiki address has the same
+        # format as a regular repo, but with ".wiki" added to the end. Whenever
+        # this repo's wiki is updated, the changes will be pushed to the repos
+        # in this list as well. This list should include this repo.
+        target: [18f/methods.wiki, 18f/ux-guide.wiki]
+
+    name: Syncronize wikis
+    runs-on: ubuntu-latest
+    steps:
+      # Rather than use a personal access token to interact with the project, we
+      # can use this GitHub App. There's an API for exchanging app credentials
+      # for a short-term token, and we use that API here.
+      - name: get token
+        uses: tibdex/github-app-token@v1
+        id: app_token
+        with:
+          app_id: ${{ secrets.PROJECT_APP_ID }}
+          installation_id: ${{ secrets.PROJECT_INSTALLATION_ID }}
+          private_key: ${{ secrets.PROJECT_PRIVATE_KEY }}
+
+      - name: checkout the source wiki
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository }}.wiki
+          token: ${{ steps.app_token.outputs.token }}
+          path: source
+
+      - name: checkout the target wiki
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ matrix.target }}
+          token: ${{ steps.app_token.outputs.token }}
+          path: target
+
+      - name: copy files
+        # These are the files within the wiki that should be synced. We assume
+        # not everything should be synced, but some content should be consistent
+        # across repos.
+        run: |
+          cp source/Automated-testing.md target/
+          cp source/Guiding-principles-and-practices.md target/
+          cp source/Releases.md target/
+          cp source/Syncing-across-repos.md target/
+          cp source/Team-roster.md target/
+          cp source/images/* target/images/
```

afaik, `cp` does not perform checksum, so `rsync` without compression would be the better tool for this:

```suggestion
          rsync -aq source/Automated-testing.md target/
          rsync -aq source/Guiding-principles-and-practices.md target/
          rsync -aq source/Releases.md target/
          rsync -aq source/Syncing-across-repos.md target/
          rsync -aq source/Team-roster.md target/
          rsync -aq source/images/ target/images/
```

You could get fancy with some type of [tool to restore mtimes](https://stackoverflow.com/questions/1964470/whats-the-equivalent-of-subversions-use-commit-times-for-git/13284229#13284229) to prevent clobbering if this workflow happened to run in both repos simultaneously, but I am doubtful that would happen in practice.

---

Can you say more about why `rsync` is better? I know it's a tool that exists and I've used it a bit, but I'm not familiar enough to say why it's better than just `cp` in this case. 😛

---

Rsync is the Swiss army knife of file synchronization in *nix based systems, but the only advantage in this case really is the one I've outlined: it is better at preventing file corruption from copying errors. If your runner's version of `cp` does this (I don't think any do, but) then there is no reason to use `rsync`.

(Rsync will normally also be faster, due to its partial-file transfer algorithm, but since these files are small, the difference is. . . not much.)

Rsync will check size and mtime when determining whether to copy files. Since your target is cloned _after_ your source, the files in each repo are guaranteed to have differing modification times.

The options I picked are:

```
 -q, --quiet                 suppress non-error messages
```

and `-a` which is a shortcut for:
```
 -r, --recursive             recurse into directories
 -l, --links                 copy symlinks as symlinks
 -p, --perms                 preserve permissions
 -t, --times                 preserve modification times
 -g, --group                 preserve group
 -o, --owner                 preserve owner (super-user only)
     --devices               preserve device files (super-user only)
     --specials              preserve special files
```

---

Wow, that's a really good reason to switch. I guess I just assumed that `cp` was safe, but this is an excellent explanation. Thank you! I'm going to add a link to this message in the comments too.